### ISSUE TITLE
cgo: C compiler "D:\\Program" not found. Is space symbol in CC variable bugged again ?

### ISSUE THREAD

**Issue by vault-thirteen** [golang/go#59512]
### What version of Go are you using (`go version`)?
> go version
> go1.20.3 windows/amd64

### Does this issue reproduce with the latest release?
Yes.

### What operating system and processor architecture are you using (`go env`)?
Windows 10,
Intel x86-64.

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=J:\Go\cache
set GOENV=C:\Users\***\AppData\Roaming\go\env
set GOEXE=.exe
set GOEXPERIMENT=
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=J:\Go\pkg\mod
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=J:\Go
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=D:\Program Files\Go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=D:\Program Files\Go\pkg\tool\windows_amd64
set GOVCS=
set GOVERSION=go1.20.3
set GCCGO=gccgo
set GOAMD64=v1
set AR=ar
set CC=D:\Program Files\LLVM\bin\clang.exe
set CXX=g++
set CGO_ENABLED=1
set GOMOD=J:\Go\src\test\sdl-wrapper\go.mod
set GOWORK=
set CGO_CFLAGS=-O2 -g
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-O2 -g
set CGO_FFLAGS=-O2 -g
set CGO_LDFLAGS=-O2 -g
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=. -m64 -mthreads -fno-caret-diagnostics -Qunused-arguments -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=C:\Users\***\AppData\Local\Temp\go-build***=/tmp/go-build -gno-record-gcc-switches

</pre></details>

### What did you do?
I am trying to compile a binary using CGO.
> go build .

The compiler exists at the path specified in CC variable.
Proof:
```
go env CC
D:\Program Files\LLVM\bin\clang.exe

"D:\Program Files\LLVM\bin\clang.exe" -v
clang version 16.0.0
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: D:\Program Files\LLVM\bin
```

### What did you expect to see?
Successful compilation.

### What did you see instead?
```
# runtime/cgo
cgo: C compiler "D:\\Program" not found: exec: "D:\\Program": file does not exist
```

**Comment by seankhliao** [comment]
From the fix in #41400, I believe you're expected to quote the CC path if it contains spaces

**Comment by vault-thirteen** [comment]
OK. I have found out that I have two ways to do it.

1. Golang way:
```
go env -w CC="\"D:\Program Files\LLVM\bin\clang.exe\""
go env CC
"D:\Program Files\LLVM\bin\clang.exe"
```

2. Windows way:
```
SET CC="D:\Program Files\LLVM\bin\clang.exe"

ECHO %CC%
"D:\Program Files\LLVM\bin\clang.exe"

go env CC
"D:\Program Files\LLVM\bin\clang.exe"
```
